src/host/premake.h

```diff
@@ -60,6 +60,8 @@
 #elif defined(__arm__) || defined(__thumb__) || defined(__TARGET_ARCH_ARM) || defined(__TARGET_ARCH_THUMB) || \
     defined(__ARM) || defined(_M_ARM) || defined(_M_ARM_T) || defined(__ARM_ARCH)
 #define PLATFORM_ARCHITECTURE "ARM"
+#elif defined(_M_RISCV64) || (defined(__riscv) && __riscv_xlen == 64)
+#define PLATFORM_ARCHITECTURE "RISCV64"
 #endif
```

Comment:
Wonder if adding
```
#else
#error "Unsupported platform"
```
would be better, clearer error message than:

>../../src/host/os_hostarch.c:11:27: error: ‘PLATFORM_ARCHITECTURE’ undeclared (first use in this function)
>   11 |         lua_pushstring(L, PLATFORM_ARCHITECTURE);

In addition, I think that `PLATFORM_ARCHITECTURE` is only used in os_hostarch.c
so moving that whole block there might make sense

Reply:
Maybe a touch out of scope for this PR, but I completely agree that would make the error messages much more readable.

Reply:
> Wonder if adding
> 
> ```
> #else
> #error "Unsupported platform"
> ```
> 
> would be better, clearer error message than:
> 
> > ../../src/host/os_hostarch.c:11:27: error: ‘PLATFORM_ARCHITECTURE’ undeclared (first use in this function)
> > 11 |         lua_pushstring(L, PLATFORM_ARCHITECTURE);
> 

Thanks for the advice. That's clearly better. Done in [f81003f](https://github.com/premake/premake-core/pull/2356/commits/f81003f5a83d719442afb672435b1658a12e8ca0)

> In addition, I think that `PLATFORM_ARCHITECTURE` is only used in os_hostarch.c so moving that whole block there might make sense

I think this is a little out of scope for this PR. I will left this for maintainers to do.

Reply:
The `#error` breaks the Windows builds when compiling the `.rc` file, you'll need to either remove that block or change the `#else` to check that [`RC_INVOKED`](https://learn.microsoft.com/en-us/windows/win32/menurc/predefined-macros) is not defined.

Reply:
> The `#error` breaks the Windows builds when compiling the `.rc` file, you'll need to either remove that block or change the `#else` to check that [`RC_INVOKED`](https://learn.microsoft.com/en-us/windows/win32/menurc/predefined-macros) is not defined.

Thanks for the pointer. I have updated the branch to check `RC_INVOKED`.

Reply:
Thanks, that looks like it worked.